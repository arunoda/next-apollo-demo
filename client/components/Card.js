import styles from './Card.module.css'

const Component = props => {
  const user = props.user
  return (
    <div className={styles.card}>
      <h3 className={styles.card__name}>{user.name}</h3>
      <div className={styles.card__email}><strong>Email</strong>: {user.email}</div>
      <div className={styles.card__address}><strong>Address</strong>: {user.address}</div>
      <div className={styles.card__phone}><strong>Phone</strong>: {user.phone}</div>
    </div>
  )
}

export default Component
