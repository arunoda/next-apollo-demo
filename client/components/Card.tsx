import React from 'react';
import styled from 'styled-components';
import { Avatar } from '@material-ui/core'
import { HomeOutlined, PhoneIphoneOutlined, EmailOutlined } from '@material-ui/icons'
import { Contact } from '../generated/graphql';

interface Props {
    data: Contact
}

const Card = ({ data }: Props) => {
    return (
        <CardContainer>
            <CardHead>
                <CardAvatar>{data.name.split("")[0]}</CardAvatar>
                <h3>{data.name}</h3>
            </CardHead>
            <h4><EmailOutlined /><span>{data.email}</span></h4>
            <h4><PhoneIphoneOutlined /><span>{data.phone}</span></h4>
            <h4><HomeOutlined /><span>{data.address}</span></h4>
        </CardContainer>
    )
}

const CardContainer = styled.div`
    width:auto;
    background-color:#222831;
    height:250px;
    border-radius:5px;
    display:flex;
    flex-direction:column;
    box-shadow:rgba(0,0,0,7);
    color:white;
    padding:1.25rem;
    @media (max-width:1310px){
        max-width:100%;
    }
    > .MuiAvatar-root{
        display:flex;
        align-items:center;
    }
    >h4{
        display:flex;
        font-size:16px;
        margin: 0.5rem 0;
        > span {
            padding:3px 0 0 10px;
            word-break: break-all;
            font-weight:400;
        }
    }
`
const CardHead = styled.div`
    display:flex;
    justify-content:center;
    flex-direction:column;
    align-items:center;
    font-size:17px;
`
const CardAvatar = styled(Avatar)`
    display:flex;
    justify-content:center;
`
export default Card;
