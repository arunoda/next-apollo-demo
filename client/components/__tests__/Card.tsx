import Card from '../Card';
import { cleanup,render } from '@testing-library/react';

describe("Card Component",()=>{
        afterEach(cleanup);
        describe("display Card",()=>{
            test("it should display card with name and match snapshot",async()=>{
                const {container}= render(
                    <Card data={{
                            name: "Nicolette Kris",
                            address: "687 Hanna Knolls Apt. 575\nNicolebury, FL 63262",
                            phone: "721-465-6539",
                            email: "Clementine_Crist@Hayes.com"
                    }} />
                    )
                    expect(container).toMatchSnapshot();
                    const text = container.getElementsByTagName('h3')[0].textContent;
                    expect(text).toBe('Nicolette Kris')
            })
        })
})