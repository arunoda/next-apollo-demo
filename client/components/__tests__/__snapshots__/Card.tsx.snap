// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component display Card it should display card with name and match snapshot 1`] = `
<div>
  <div
    class="Card__CardContainer-sc-1b3jwt6-0 fpjUII"
  >
    <div
      class="Card__CardHead-sc-1b3jwt6-1 iPUqjq"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle Card__CardAvatar-sc-1b3jwt6-2 QmZGE MuiAvatar-colorDefault"
      >
        N
      </div>
      <h3>
        Nicolette Kris
      </h3>
    </div>
    <h4>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
        />
      </svg>
      <span>
        Clementine_Crist@Hayes.com
      </span>
    </h4>
    <h4>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 1h-8C6.12 1 5 2.12 5 3.5v17C5 21.88 6.12 23 7.5 23h8c1.38 0 2.5-1.12 2.5-2.5v-17C18 2.12 16.88 1 15.5 1zm-4 21c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm4.5-4H7V4h9v14z"
        />
      </svg>
      <span>
        721-465-6539
      </span>
    </h4>
    <h4>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.69l5 4.5V18h-2v-6H9v6H7v-7.81l5-4.5M12 3L2 12h3v8h6v-6h2v6h6v-8h3L12 3z"
        />
      </svg>
      <span>
        687 Hanna Knolls Apt. 575
Nicolebury, FL 63262
      </span>
    </h4>
  </div>
</div>
`;
